text/0268-acceptance-testing-refactor.md

```diff
@@ -0,0 +1,223 @@
+  * setup `this.triggerKeyEvent` as an async helper to trigger a `KeyEvent` on the specified selector
+  * setup `this.triggerEvent` as an async helper to trigger an event on the specified selector
+  * setup `this.fillIn` as an async helper to enter text on the specified selector
+  * setup `this.waitUntil` as an async helper returning a promise which resolves when the provided callback returns a truthy value
+  * setup `this.focus` as an async helper which focuses the specified element
+  * setup `this.blur` as an async helper which unfocuses the specified element
+
+### `setupRenderingTest`
+
+The `setupRenderingTest` function proposed in
+[emberjs/rfcs#232](https://github.com/emberjs/rfcs/blob/master/text/0232-simplify-qunit-testing-api.md)
+(and implemented in
+[ember-qunit](https://github.com/emberjs/ember-qunit)@3.0.0) will be modified to add the same DOM interaction helpers:
+
+* setup a getter for `this.element` which returns the DOM element representing
+  the root element
+* if `jQuery` is present in the application sets up `this.$` method to run
+  jQuery selectors rooted to `this.element`
+* setup `this.click` as an async helper to click on the specified selector
+* setup `this.tap` as an async helper to tap on the specified selector
+* setup `this.triggerKeyEvent` as an async helper to trigger a `KeyEvent` on the specified selector
+* setup `this.triggerEvent` as an async helper to trigger an event on the specified selector
+* setup `this.fillIn` as an async helper to enter text on the specified selector
+* setup `this.waitUntil` as an async helper returning a promise which resolves when the provided callback returns a truthy value
+* setup `this.focus` as an async helper which focuses the specified element
+* setup `this.blur` as an async helper which unfocuses the specified element
+
+Once implemented, `setupRenderingTest` and `setupAcceptanceTest` will diverge from each other in very few ways.
+
+### Importable helpers
+
+Since the design of this system relies on both the test helpers being applied
+to the test context **and** the usage of `async` / `await`, a few importable
+helpers are being introduced to help avoid extra noise (e.g. "rightward shift")
+in tests.
+
+This means that users will be able to use either of the following interchangably:
+
+```js
+// ***** importable helper *****
+import { click } from 'ember-test-helpers';
+
+// ...snip...
+test('does something', async function(assert) {
+  // ...snip...
+  await click('.selector-here');
+  // ...snip...
+});
+
+// ***** test context helper *****
+
+// ...snip...
+test('does something', async function(assert) {
+  // ...snip...
+  await this.click('.selector-here');
+  // ...snip...
+});
+```
+
+## Changes from Current System
+
+Here is a brief list of the more important but possibly understated changes
+being proposed here:
+
+* The global test helpers that exist now, will no longer be present (e.g.
```

Comment:
Really not present, or rather deprecated? So they can co-exist, see the following section.

Reply:
When running tests with the old `moduleForAcceptance` API they will be present, this section is referring to tests using `setupAcceptanceTest(...)` API and is correct in saying that the global methods _will not_ be present.

Reply:
That makes sense! Maybe this should be stated to prevent misunderstandings like I had?